test/simulator/main/org/apache/cassandra/simulator/systems/SimulatedAction.java

```diff
@@ -382,8 +382,11 @@ List<Action> applyToMessage(IInvokableInstance from, IInvokableInstance to, IMes
                     notify = from;
                 }
                 boolean isTimeout = deliver != FAILURE;
+                Executor notifierExecutor = notify.executorFor(verb.id);
+                if (notifierExecutor instanceof ImmediateExecutor)
```

Comment:
Why is this unwrapping necessary? Worth a comment?

Reply:
its the same as the above logic.  If any stage is `ImmediateExecutor` then the executor isn't changed out during simulation... so the logic that requires a cast will cause a ClassCastException